proxy/src/main.rs

```diff
@@ -181,49 +188,42 @@ fn main() -> Result<(), Box<dyn Error>> {
-                                                let received_task =
-                                                    Task::deserialise(msg.data).unwrap();
-                                                let add_res: RedisResult<u32> = connection.sadd(
-                                                    args.value_of("redis-set").unwrap(),
-                                                    &received_task,
-                                                );
-                                                if let Ok(_s) = add_res {
-                                                    channel.basic_ack(msg.delivery_tag, false)
-                                                } else {
-                                                    channel.basic_reject(
-                                                        msg.delivery_tag,
-                                                        BasicRejectOptions::default(), //TODO
-                                                    )
-                                                }
-                                            })
-                                        })
-                                })
-                        })
-                    }),
-                )
-                    .wait_future()
-                    .expect("Could not connect to rabbitMQ");
+                let client = Client::connect(&rmq_addr, ConnectionProperties::default()).wait()?;
+                // Finds collection and sees the tasks
+                let channel = client.create_channel().wait()?;
+                let queue = channel
+                    .queue_declare(
+                        args.value_of("rabbitmq-collection-queue").unwrap(),
+                        QueueDeclareOptions::default(),
+                        FieldTable::default(),
+                    ).wait()?;
+
+                let consumer = channel
+                    .basic_consume(
+                        &queue,
+                        args.value_of("rabbitmq-consumer-tag").unwrap(),
+                        BasicConsumeOptions::default(),
+                        FieldTable::default(),
+                    ).wait()?;
+
+
+                // Copies every task from collection to redis
+                consumer.for_each(move |msg| {
+                    let received_task =
+                        Task::deserialise(msg.data).unwrap();
+                    let add_res: RedisResult<u32> = connection.sadd(
+                        args.value_of("redis-set").unwrap(),
+                        &received_task,
+                    );
+                    if let Ok(_s) = add_res {
+                        channel.basic_ack(msg.delivery_tag, false)
+                    } else {
+                        channel.basic_reject(
+                            msg.delivery_tag,
+                            BasicRejectOptions { requeue: true },
+                        )
+                    }
+                }).wait().unwrap();
```

Comment:
```suggestion
                }).wait()?;
```
Could this work instead?